helper/lorawan-mac-helper.cc

```diff
@@ -461,7 +461,7 @@ LorawanMacHelper::SetSpreadingFactorsUp(NodeContainer endDevices,
     NS_LOG_FUNCTION_NOARGS();
 
     std::vector<int> sfQuantity(7, 0);
-    for (NodeContainer::Iterator j = endDevices.Begin(); j != endDevices.End(); ++j)
+    for (auto j = endDevices.Begin(); j != endDevices.End(); ++j)
```

Comment:
You can reduce all these iterator even more using range-base: for(auto object : list).
For example:
```cpp
for (const auto& device : endDevices)
{
   Ptr<MobilityModel> position = device->GetObject<MobilityModel>();
}
```
If its too much hasle to change the existing for, please ignore my comment, but take it into consideration for new additons.

Reply:
Agree with @albertogara, I suggest changing this

Reply:
yeap, that's why I mentioned.... consider it for "future additions" of newly introduced fors.
overall, iterators are frowned upon in the current ns-3 style. for-range loops are not mandatory but they are preferred.

Reply:
@albertogara if I try to loop on NodeContainer as you propose I get `this range-based 'for' statement requires a suitable "begin" function and none was found`. Are you sure range-based loops work with ns-3's Container classes?

Reply:
You're right, it actually doesn't work with NodeContainers

Reply:
Yup, my bad. It only work on collections (list, vector, maps)